#3: Prepare release of EVSCL (#5)
exasol/exasol-virtual-schema-common-lua: doc/system_requirements.md

# System Requirement Specification Exasol Virtual Schema Common for Lua

## Introduction

> Exasol Virtual Schema Common for Lua (short "EVSCL") is base library for [Virtual Schemas](https://docs.exasol.com/db/latest/database_concepts/virtual_schemas.htm) that connect to Exasol databases. 

Reviewer: ```suggestion
Exasol Virtual Schema Common for Lua (short "EVSCL") is a base library for [Virtual Schemas](https://docs.exasol.com/db/latest/database_concepts/virtual_schemas.htm) that connect to Exasol databases. 
```

Virtual Schemas are conceptually related to database views. The main difference compared to database views is that Virtual Schema sources can be almost any data source. To make this possible, you need a so-called "Virtual Schema adapter", an Exasol extension that contains the logic to translate between the source and the projection provided by the Virtual Schema.

From an end-user's perspective a Virtual Schema looks and feels like an internal schema of Exasol &mdash; albeit one that can only be read.

## About This Document

### Target Audience

> The target audience are end-users, requirement engineers, software designers and quality assurance. See section ["Stakeholders"](#stakeholders) for more details.

Reviewer: requirement engineers, software designers and quality assurance are not listed in the "Stakeholder" section

Author: Added.

* Schema: database schema. In Exasol the top-level grouping element for tables and scripts in a database.
* Virtual Schema: projection of data from a data source that - from the end-user's perspective - looks like a regular schema.

In the following subsections central terms are explained in more detail.

## Features

Features are the highest level requirements in this document that describe the main functionality of EVSL.

Since features are defined in the actual Virtual Schemas that depend on this library, none are listed here. We intentionally start the requirement chain with the next level.

## Functional Requirements

### Source Metadata Scan

The main work when creating a new virtual schema is to map the structure and data types of the data source to virtual schema metadata. This happens by scanning the data source and then reporting table and column structure back. This section holds requirements that are common for scanning an Exasol data source. 

#### Reading Table Metadata From a Schema
`req~reading-table-metadata-from-a-schema~1`

EVSCL reads the list of tables contained in the source database schema.

Needs: dsn

> #### Reading Columns Metadata From a Table

Reviewer: ```suggestion
#### Reading Column Metadata From a Table
```

`req~reading-column-metadata-from-a-table~1`

> EVSCL reads the list of columns and their attributes contained from a table in the source database.

Reviewer: ```suggestion
EVSCL reads the list of columns and their attributes from a table in the source database.
```

Needs: dsn

#### Include Tables
`req~include-tables~1`

EVSCL offers an interface to specify the tables from the source database schema that should be included in the virtual schema. 

Needs: dsn

### Shared Adapter Properties of Virtual Schemas That Access an Exasol Database

> Adapter properties allow users to configure their virtual schemas. In this section we define properties that all virtual schemas share which access an Exasol data source. 

Reviewer: ```suggestion
Adapter properties allow VS Owners to configure their virtual schemas. In this section we define properties that all virtual schemas share which access an Exasol data source. 
```
